### PR TITLE
Updates regex expression to match case insensitive exact org name

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
Updates org filter regex expression. Details on the JIRA ticket.
_Issue:_
Due to the regex matching while filtering courses (https://github.com/edly-io/edx-platform/blob/0680d4492dc0c1a7da662560577c6bec7c9e2398/openedx/core/djangoapps/content/course_overviews/models.py#L579), Trial filters courses including those of Trial1 and Trial2. This means that any org name where one org has a same name w.r.t to another with case changed or the org name is a substring to another organization name, courses of that org would also be included.
image

 _Fix:_
This issue has been fixed by cherry-picking the following commit: https://github.com/edx/edx-platform/commit/99960d7001f2378c2918e60a5a1f5b9de621fc68

The new query is:
image

**Related edX PR:** https://github.com/edx/edx-platform/pull/14471/

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-2126

### CODE REVIEW

**Review by ziafazal — APPROVED** [review]
👍

File: openedx/core/djangoapps/content/course_overviews/models.py [hunk 1]
```diff
@@ -576,7 +577,10 @@ def get_all_courses(cls, orgs=None, filter_=None):
             # In rare cases, courses belonging to the same org may be accidentally assigned
             # an org code with a different casing (e.g., Harvardx as opposed to HarvardX).
             # Case-insensitive matching allows us to deal with this kind of dirty data.
-            course_overviews = course_overviews.filter(org__iregex=r'(' + '|'.join(orgs) + ')')
+            org_filter = Q()  # Avoiding the `reduce()` for more readability, so a no-op filter starter is needed.
+            for org in orgs:
+                org_filter |= Q(org__iexact=org)
+            course_overviews = course_overviews.filter(org_filter)
```
Comment (zubair-arbi) [inline openedx/core/djangoapps/content/course_overviews/models.py:583]: nit: add a new line before de-indent.

File: openedx/core/djangoapps/content/course_overviews/models.py [hunk 2]
```diff
@@ -7,7 +7,8 @@
 
 from django.conf import settings
 from django.db import models, transaction
-from django.db.models.fields import BooleanField, DateTimeField, DecimalField, TextField, FloatField, IntegerField
```
Comment (zubair-arbi) [inline openedx/core/djangoapps/content/course_overviews/models.py:10]: Remove the change in this line to reduce git diff.

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
Some small nits, otherwise changes looks good.

Please add the details on your PR as well. also mention the new query.

File: openedx/core/djangoapps/content/course_overviews/models.py (hunk 2 above)
Comment (rehan99000) [reply openedx/core/djangoapps/content/course_overviews/models.py:10]: done

File: openedx/core/djangoapps/content/course_overviews/models.py (hunk 1 above)
Comment (rehan99000) [reply openedx/core/djangoapps/content/course_overviews/models.py:584]: done

**Review by zubair-arbi — APPROVED** [review]
Changes looks good now 👍 
Please squash your commits into one commit.

**Pull request merged**